registry/config_etcd.go

```diff
@@ -0,0 +1,150 @@
+package registry
+
+import (
+	"fmt"
+	"github.com/cloudogu/cesapp-lib/core"
+	"github.com/coreos/etcd/client"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+type etcdConfigurationContext struct {
+	parent string
+	client *resilentEtcdClient
+}
+
+// Set sets a configuration value in current context
+func (ecc *etcdConfigurationContext) Set(key, value string) error {
+	return ecc.set(key, value, nil)
+}
+
+// SetWithLifetime sets a configuration value in current context with the given lifetime
+func (ecc *etcdConfigurationContext) SetWithLifetime(key, value string, timeToLiveInSeconds int) error {
+	duration, err := time.ParseDuration(fmt.Sprintf("%ds", timeToLiveInSeconds))
+	if err != nil {
+		return errors.Wrapf(err, "could not create the curation '%d'", timeToLiveInSeconds)
+	}
+	return ecc.set(key, value, &client.SetOptions{
+		TTL: duration,
+	})
+}
+
+// Refresh will refresh the ttl of a key
+func (ecc *etcdConfigurationContext) Refresh(key string, timeToLiveInSeconds int) error {
+	duration, err := time.ParseDuration(fmt.Sprintf("%ds", timeToLiveInSeconds))
+	if err != nil {
+		return errors.Wrapf(err, "could not create the curation '%d'", timeToLiveInSeconds)
+	}
+
+	err = ecc.set(key, "", &client.SetOptions{
+		TTL:     duration,
+		Refresh: true,
+	})
+	return err
+}
+
+func (ecc *etcdConfigurationContext) set(key, value string, options *client.SetOptions) error {
+	path := ecc.parent + "/" + key
+	core.GetLogger().Debug("try to set config key", path)
+
+	core.GetLogger().Debugf("write key %s", key)
```

Comment:
Reading the debug output for registry interaction is like trying to drink the ocean. Should we merge these two lines in favour of a combined output?

Reply:
Done